kpm/lib/kpm/nexus_helper/nexus_api_calls_v2.rb

```diff
@@ -0,0 +1,156 @@
+        file_name = get_file_name(coordinates)
+        destination = File.join(File.expand_path(destination || "."), file_name)
+        http_client = get_client
+        query_params = get_query_params(coordinates)
+        url = get_url(PULL_ARTIFACT_ENDPOINT, configuration)
+        response = http_client.get(url,query_params)
+
+        case response.status
+          when 301, 307
+            # Follow redirect and stream in chunks.
+            File.open(destination, "wb") do |io|
+              http_client.get(response.content.gsub(/If you are not automatically redirected use this url: /, "")) do |chunk|
+                io.write(chunk)
+              end
+            end
+          when 404
+            raise StandardError.new('The artifact you requested information for could not be found. Please ensure it exists inside the Nexus.')
+          else
+            raise UnexpectedStatusCodeException.new(response.status)
+        end
+        {
+            :file_name => file_name,
+            :file_path => File.expand_path(destination),
+            :version   => version,
+            :size      => File.size(File.expand_path(destination))
+        }
+      end
+
+      private
+        def parse_coordinates(coordinates)
+          split_coordinates = coordinates.split(":")
+          if(split_coordinates.size < 3 or split_coordinates.size > 5)
+            raise ArtifactMalformedException
+          end
+
+          artifact = Hash.new
+
+          artifact[:group_id] = split_coordinates[0]
+          artifact[:artifact_id] = split_coordinates[1]
+          artifact[:extension] = split_coordinates.size > 3 ? split_coordinates[2] : "jar"
+          artifact[:classifier] = split_coordinates.size > 4 ? split_coordinates[3] : nil
+          artifact[:version] = split_coordinates[-1]
+
+          artifact[:version].upcase! if version == "latest"
+
+          return artifact
+        end
+
+        def get_file_name(coordinates)
+          artifact = parse_coordinates(coordinates)
+
+          if artifact[:version].casecmp("latest")
+             artifact[:version] = REXML::Document.new(get_artifact_info(coordinates)).elements["//version"].text
+          end
+
+          if artifact[:classifier].nil?
+            "#{artifact[:artifact_id]}-#{artifact[:version]}.#{artifact[:extension]}"
+          else
+            "#{artifact[:artifact_id]}-#{artifact[:version]}-#{artifact[:classifier]}.#{artifact[:extension]}"
+          end
+        end
+
+        def get_query_params(coordinates)
+          artifact = parse_coordinates(coordinates)
+          @version = artifact[:version]
+          query = {:g => artifact[:group_id], :a => artifact[:artifact_id], :e => artifact[:extension], :v => version, :r => configuration[:repository]}
+          query.merge!({:c => artifact[:classifier]}) unless artifact[:classifier].nil?
+          query
+        end
+
+        def get_client
+          client = HTTPClient.new
```

Comment:
How hard would it be to simply use `net/http` instead? That way, we wouldn't have to add another dependency in the `gemspec` (ideally, we would only depend on the killbill client, and maybe things like `thor` for the command line).